comp/otelcol/collector-contrib/impl/components.go

```diff
@@ -69,6 +70,15 @@ func components() (otelcol.Factories, error) {
 	if err != nil {
 		return otelcol.Factories{}, err
 	}
+	factories.ExtensionModules = make(map[component.Type]string, len(factories.Extensions))
+	factories.ExtensionModules[zpagesextension.NewFactory().Type()] = "go.opentelemetry.io/collector/extension/zpagesextension v0.111.0"
+	factories.ExtensionModules[healthcheckextension.NewFactory().Type()] = "github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension v0.111.0"
+	factories.ExtensionModules[pprofextension.NewFactory().Type()] = "github.com/open-telemetry/opentelemetry-collector-contrib/extension/pprofextension v0.111.0"
+	factories.ExtensionModules[dockerobserver.NewFactory().Type()] = "github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer/dockerobserver v0.111.0"
+	factories.ExtensionModules[ecsobserver.NewFactory().Type()] = "github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer/ecsobserver v0.111.0"
```

Comment:
Hello @dineshg13 !

I’ve updated the `manifest.yaml` file and ran `inv collector.generate` in https://github.com/DataDog/datadog-agent/pull/30061/commits/ea3597e4d592941910a8520546375838d6db3bae

Doing so broke the `docker_image_build_otel` job https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/687482408#L1608 because of an incompatibility between the OTEL version used in this PR (`0.111.0`) and the OTEL version (`0.104.0`) used in the branch (`7.57.x-otel-beta-v1`) hard-coded in this job. https://github.com/DataDog/datadog-agent/blob/32e6f2692b8520f0b7d7f42cf7a16d7accd473fc/Dockerfiles/agent-ot/Dockerfile.agent-otel#L2

So, I had to fix it.

IMHO, re-cloning the `datadog-agent` repo https://github.com/DataDog/datadog-agent/blob/32e6f2692b8520f0b7d7f42cf7a16d7accd473fc/Dockerfiles/agent-ot/Dockerfile.agent-otel#L29 looks like a bad idea. We’d rather rely on the working tree already checked out by GitLab at the proper commit.
But I’ll let that for another dedicated PR.

Reply:
It is looks like a bug in the test.  @jackgopack4 can you please take a look.

Reply:
I'm likely going to approach this by adding a build arg to dockerfile to allow pulling in local files instead of github when running in the test pipeline

Reply:
Have a PR: https://github.com/DataDog/datadog-agent/pull/30619
with one commit: [`f13f4b3` (#30619)](https://github.com/DataDog/datadog-agent/pull/30619/commits/f13f4b3b13ccf5af52e4eafb11bdd81b69825d7f)
That should fix this; just waiting on the draft pipelines to run to ensure it doesn't break anything else